src/main/java/net/dv8tion/jda/internal/requests/restaction/ChannelActionImpl.java

```diff
@@ -88,6 +99,13 @@ public ChannelActionImpl(Class<T> clazz, String name, Guild guild, ChannelType t
         this.name = name;
     }
 
+    @Nonnull
+    @Override
+    public ChannelActionImpl<T> reason(@Nullable String reason)
+    {
+        return (ChannelActionImpl<T>) super.reason(reason);
+    }
```

Comment:
Can AuditableRestActionImpl be made smart enough to not need this? or will we always need this?

Reply:
That would require rewriting all classes using it, which seems out of scope for this PR.